core/spec/models/spree/promotion/actions/create_item_adjustments_spec.rb

```diff
@@ -4,11 +4,11 @@ module Spree
   class Promotion
     module Actions
       describe CreateItemAdjustments, type: :model do
-        let(:order) { create(:order) }
+        let(:order) { create(:order_with_line_items, line_items_count: 1) }
         let(:promotion) { create(:promotion, :with_line_item_adjustment, adjustment_rate: adjustment_amount) }
         let(:adjustment_amount) { 10 }
         let(:action) { promotion.actions.first! }
-        let!(:line_item) { create(:line_item, order: order) }
+        let(:line_item) { order.line_items.to_a.first }
```

Comment:
@gmacdougall this ensures that `line_item` refers to the same object that `OrderUpdater` and the promo code & etc will modify.  Our current implementation of the order factory happens to preload `order.line_items` properly, but that's not tested and I'm not sure it was intentional either.  Perhaps I could remove it here but make an additional PR that tests this explicitly in `order_factory_spec`?